core/controllers/topic_viewer.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the topics viewer page."""
+
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import story_services
+from core.domain import topic_services
+import feconf
+
+
+class TopicPageDataHandler(base.BaseHandler):
+    """Manages the data that needs to be displayed to a learner on the topic
+    viewer page.
+    """
+
+    @acl_decorators.can_access_topic_viewer_page
+    def get(self, topic_id):
+        """Handles GET requests."""
+
+        topic = topic_services.get_topic_by_id(topic_id)
```

Comment:
I think you need the ENABLE_NEW_STRUCTURES check here?

Reply:
This will be triggered only when an API call is done from the topic viewer page which has this, so that's why I didn't add  the check. Will do

Reply:
Done.